### PR TITLE
feat: Added temporal codec server route & cookbooks. Updated the CONTRIBUTING.md file 

### PR DESCRIPTION
**Pull request by Vedantsahai18**
> [!IMPORTANT]
> Added `/temporal/decode` route, updated setup instructions in `CONTRIBUTING.md`, and modified Docker and FastAPI configurations.
> 
>   - **New Features**:
>     - Added `/temporal/decode` route in `router.py` for decoding temporal payloads using `PydanticEncodingPayloadConverter`.
>   - **Documentation**:
>     - Updated `CONTRIBUTING.md` with setup instructions, including environment setup, Docker volume creation, and running the project in single or multi-tenant mode.
>   - **Configuration**:
>     - Added `internal.router` to `web.py` to include the new internal routes.
>     - Updated `docker-compose.yml` to expose port `8080` for `agents-api` and `agents-api-multi-tenant` services.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=julep-ai%2Fjulep&utm_source=github&utm_medium=referral)<sup> for e8238b27e4aa487c49d1e71711c7424f00d2ddd4. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -30,44 +30,76 @@ Improvements to documentation are always appreciated! If you see areas that coul
...
+```
+
+> **Note:** In multi-tenant mode, you need to generate a JWT token that act as an API KEY to interact with the SDK in the local setup.
+
+##### 6. Generate a JWT Token (Only for Multi-Tenant Mode)
+
+To generate a JWT token, `jwt-cli` is required. Install it using the following command:
+
+```bash
+npm install -g jwt-cli
```
Comment (creatorrr) [inline CONTRIBUTING.md:87]: no the lib is different: `cargo install jwt-cli --locked`

File: agents-api/agents_api/routers/internal/router.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+from ...worker.codec import PydanticEncodingPayloadConverter
+
+router: APIRouter = APIRouter()
+
+converter = PydanticEncodingPayloadConverter() 
+
+# Decode route
+@router.post("/temporal/decode", tags=["temporal"])
+async def decode_payloads(req: Request)-> Response:
```
[the remark this answers is not in the document]
Comment (creatorrr) [reply agents-api/agents_api/routers/internal/router.py:13]: we should change return type to `-> dict`

File: cookbooks/03_Cookbook.ipynb [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Implementation\n",
+    "\n",
+    "To recreate the notebook and see the code implementation for this task, you can access the Google Colab notebook using the link below:\n",
+    "\n",
+    "[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://drive.google.com/file/d/14ntgAPmm2-QD3KArc5-lyM5wCMyTx5b1/view?usp=drive_link)\n",
```
Comment (creatorrr) [inline cookbooks/03_Cookbook.ipynb:60]: This isn't the correct link. Use https://openincolab.com/ to generate it but make sure to change the branch name to `dev`

File: cookbooks/04_Cookbook.ipynb [hunk 4]
```diff
@@ -0,0 +1,449 @@
...
+    "# Listing all agents available in the account\n",
+    "get_agents = client.agents.list()\n",
+    "get_agents.items"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We get the current agent ID from the list of agents."
```
Comment (creatorrr) [inline cookbooks/04_Cookbook.ipynb:200]: remove the list agents, list tasks, etc. they are useless distractions

File: cookbooks/04_Cookbook.ipynb [hunk 5]
```diff
@@ -0,0 +1,449 @@
...
+   "source": [
+    "Listing the execution states."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
```
Comment (creatorrr) [inline cookbooks/04_Cookbook.ipynb:423]: mention that list transitions doesnt wait, just lists whatever transitions have happened so far

Comment (creatorrr) [reply cookbooks/04_Cookbook.ipynb:423]: and move `.list()` above `.stream()`

File: cookbooks/04_Cookbook.ipynb [hunk 6]
```diff
@@ -0,0 +1,449 @@
...
+    "execution = client.executions.get(execution.id)\n",
+    "#printing the output\n",
+    "execution.output"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Streaming execution steps of the defined Task."
```
Comment (creatorrr) [inline cookbooks/04_Cookbook.ipynb:399]: Mention that this is a blocking operation, waits for the execution to get over

**Comment by creatorrr** [comment]
Also- change names of the cookbook files to something more descriptive like `01-Sarcastic_News_Headline_Generator.ipynb`

**Comment by creatorrr** [comment]
Also run `poe check` before merging

File: agents-api/agents_api/routers/internal/router.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Vedantsahai18) [reply agents-api/agents_api/routers/internal/router.py:13]: will JSONResponse be more appropriate rather than a dict

File: CONTRIBUTING.md (hunk 1 above)
Comment (Vedantsahai18) [reply CONTRIBUTING.md:87]: for cargo installation the users might require extra set installation like Rust and Cargo. Instead can we just ask to install 'jwt-cli' depending on their platform?

**Review by HamadaSalhab — COMMENTED** [review]
It would be better if you show the final output of the task execution in the cookbooks.

**Pull request merged**